lua/session_manager/utils.lua

```diff
@@ -125,7 +125,10 @@ function utils.dir_to_session_filename(dir)
 end
 
 function utils.is_normal_buffer(buffer)
-  return #vim.api.nvim_buf_get_option(buffer, 'buftype') == 0 and vim.api.nvim_buf_get_option(buffer, 'buflisted')
+  local is_normal_buftype = #vim.api.nvim_buf_get_option(buffer, 'buftype') == 0
+  local is_buflisted = vim.api.nvim_buf_get_option(buffer, 'buflisted')
+  local is_normal_filetype = not vim.tbl_contains(config.autosave_ignore_filetypes, vim.api.nvim_buf_get_option(buffer, 'filetype'))
+  return is_normal_buftype and is_buflisted and is_normal_filetype
```

Comment:
I would like to make it more optimal since it called in a loop. Let's evaluate conditions lazily and return `false` on first mismatch:

```suggestion
  if #vim.api.nvim_buf_get_option(buffer, 'buftype') ~= 0 then
      return false
  end
  if not vim.api.nvim_buf_get_option(buffer, 'buflisted') then
      return false
  end
  if vim.tbl_contains(config.autosave_ignore_filetypes, vim.api.nvim_buf_get_option(buffer, 'filetype')) then
      return false
  return true
```